packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-suggestions.tsx

```diff
@@ -0,0 +1,257 @@
+
+  public initialize() {
+    this.smartSnippetQuestionsList = buildSmartSnippetQuestionsList(
+      this.bindings.engine
+    );
+  }
+
+  private get style() {
+    const styleTag = this.host
+      .querySelector('template')
+      ?.content.querySelector('style');
+    if (!styleTag) {
+      return this.snippetStyle;
+    }
+    return styleTag.innerHTML;
+  }
+
+  private getRelatedQuestionId(relatedQuestion: SmartSnippetRelatedQuestion) {
+    // TODO: Change to the snippet's unique ID
+    return `${this.id}-${relatedQuestion.question
+      .toLowerCase()
+      .replace(/\s/g, '-')
+      .replace(/[^a-z-]/g, '')}`;
+  }
+
+  private getQuestionPart(
+    prefix: string,
+    relatedQuestion: SmartSnippetRelatedQuestion
+  ) {
+    return prefix + (relatedQuestion.expanded ? '-expanded' : '-collapsed');
+  }
+
+  private toggleQuestion(relatedQuestion: SmartSnippetRelatedQuestion) {
+    if (relatedQuestion.expanded) {
+      this.smartSnippetQuestionsList.collapse(relatedQuestion.documentId);
+    } else {
+      this.smartSnippetQuestionsList.expand(relatedQuestion.documentId);
+    }
+  }
+
+  private renderQuestion(relatedQuestion: SmartSnippetRelatedQuestion) {
+    return (
+      <Button
+        style="text-neutral"
+        part={this.getQuestionPart('question-button', relatedQuestion)}
+        onClick={() => this.toggleQuestion(relatedQuestion)}
+        class="flex items-center px-4"
+        ariaExpanded={`${relatedQuestion.expanded}`}
+        ariaControls={this.getRelatedQuestionId(relatedQuestion)}
+      >
+        <atomic-icon
+          icon={relatedQuestion.expanded ? ArrowDown : ArrowRight}
+          part={this.getQuestionPart('question-icon', relatedQuestion)}
+          class="w-2.5 mr-3 stroke-[1.25]"
+        ></atomic-icon>
+        <Heading
+          level={this.headingLevel ? this.headingLevel + 1 : this.headingLevel}
+          class="text-left text-xl font-bold py-4"
+          part={this.getQuestionPart('question-text', relatedQuestion)}
+        >
+          {relatedQuestion.question}
+        </Heading>
+      </Button>
+    );
+  }
+
+  private renderContent(relatedQuestion: SmartSnippetRelatedQuestion) {
+    return (
+      <atomic-smart-snippet-answer
+        exportparts="answer"
+        htmlContent={relatedQuestion.answer}
+        innerStyle={this.style}
+      ></atomic-smart-snippet-answer>
+    );
+  }
+
+  private renderSource(relatedQuestion: SmartSnippetRelatedQuestion) {
+    const {source} = relatedQuestion;
+    if (!source) {
+      return [];
+    }
+    const interactiveResult = buildInteractiveResult(this.bindings.engine!, {
+      options: {result: source},
+    });
+    return (
+      <footer
+        part="footer"
+        aria-label={this.bindings.i18n.t('smart-snippet-source')}
+      >
+        <LinkWithResultAnalytics
+          title={source.clickUri}
+          href={source.clickUri}
+          target="_self"
```

Comment:
Why is the target forced to self ? It's because it's required in LinkWithResultAnalytics ?

If so, we will need to make this configurable for everything that's a result link. That was used quite a lot in JSUI.

For example, people want to configure all links in the page to always open in a new window/tab. And so I don't think it should be a "per component" option, but a "per interface" global setting.

This should be another story though.

Reply:
I copy and pasted this from smart snippet. I think it should be fixed along with highlighting by Simo